src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,63 @@
+.calendar {
+  /* Positioning */
+  position: relative;
+
+  /* Display & Box Model */
+  display: flex;
+  margin-inline: auto;
+  padding: 30px 10px 10px;
+  width: $calendar-width;
+  flex-wrap: wrap;
+
+  /* Other */
+  gap: $gap;
+
+  &__day {
+    /* Display & Box Model */
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid #000;
+
+    /* Other */
+    background-color: #eee;
+    transition: transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  &--month {
+    @for $num from 28 through 31 {
+      &-#{$num} {
+        .calendar__day:nth-child(n + #{$num + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--start {
+    @each $day, $num in $week {
+      &-#{$day} {
+        .calendar__day:first-child {
+          margin-left: ($day-size + $gap) * $num;
+        }
+      }
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .day-#{$i}::before {
+    content: "#{$i}";
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
```

Comment:
Don't use SASS loops for styles that stay the same for all elements of the group